From: Aurelien Aptel <aaptel@suse.com>
Subject: [PATCH] cifs: add NULL check for ses->tcon_ipc
Git-commit: 59463eb88829f646aed13283fd84d02a475334fe
References: bsc#1178270
Patch-mainline: v5.10-rc7

In some scenarios (DFS and BAD_NETWORK_NAME) set_root_set() can be
called with a NULL ses->tcon_ipc.

Acked-by: Aurelien Aptel <aaptel@suse.com>
Reviewed-by: Aurelien Aptel <aaptel@suse.com>
Signed-off-by: Aurelien Aptel <aaptel@suse.com>

---
 fs/cifs/connect.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/fs/cifs/connect.c
+++ b/fs/cifs/connect.c
@@ -4677,7 +4677,8 @@ static void set_root_ses(struct cifs_sb_
 	if (ses) {
 		spin_lock(&cifs_tcp_ses_lock);
 		ses->ses_count++;
-		ses->tcon_ipc->remap = cifs_remap(cifs_sb);
+		if (ses->tcon_ipc)
+			ses->tcon_ipc->remap = cifs_remap(cifs_sb);
 		spin_unlock(&cifs_tcp_ses_lock);
 	}
 	*root_ses = ses;
